pkg/utils/kubernetes/kubernetes.go

```diff
@@ -421,11 +418,7 @@ func OwnedBy(obj runtime.Object, apiVersion, kind, name string, uid types.UID) b
 // is provided then it will be applied for each object right after listing all objects. If no object remains then nil
 // is returned. The Items field in the list object will be populated with the result returned from the server after
 // applying the filter function (if provided).
-func NewestObject(ctx context.Context, c client.Client, listObj runtime.Object, filterFn func(runtime.Object) bool, listOpts ...client.ListOption) (runtime.Object, error) {
-	if !meta.IsListType(listObj) {
-		return nil, fmt.Errorf("provided <listObj> is not a List type")
-	}
-
+func NewestObject(ctx context.Context, c client.Client, listObj client.ObjectList, filterFn func(runtime.Object) bool, listOpts ...client.ListOption) (runtime.Object, error) {
```

Comment:
Hmm, `meta.EachListItem` only supports `runtime.Object` so we would need to type-assert there again before calling the filter.
So I think, this should be fine for now, as it's not strictly required to get the vendoring through.
I have an open task in the umbrella issue about getting rid of the remaining `meta.Accessor` usages and this would be covered by this.

Reply:
OK, makes sense, thanks.